src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -39,14 +52,30 @@ class RustHighlightingAnnotator : Annotator {
             }
         }
 
-        override fun visitPath(o: RustPathElement) {
-            o.reference.resolve().let {
-                if (it is RustPatBindingElement && it.isMut) {
-                    holder.highlight(o.identifier, RustColor.MUT_BINDING)
-                }
+        // @TODO ReferenceElement
+        override fun visitPath(path: RustPathElement) {
+            val ref = path.reference.resolve() ?: return
+
+            if (ref is RustPathElement) {
+                // Not sure how a path references itself, this causes overflow otherwise.
+                return
             }
+            // This highlights the *path* dependent on the reference type.
+            resolveColor(ref)?.let { holder.highlight(path.identifier, it) }
         }
 
+        override fun visitMethodCallExpr(o: RustMethodCallExprElement) {
+            holder.highlight(o.identifier, RustColor.INSTANCE_METHOD)
+        }
+
+        // @TODO Bind types and paths similarly
+        //       grammar was tough to read, doesn't look like it can differentiate between bottom types.
```

Comment:
Grammar won't tell you what kind of item does the path point to. You have to resolve it, which should be already doable.